service/src/main/java/uk/nhs/adaptors/gp2gp/TransformJsonToXml.java

```diff
@@ -0,0 +1,225 @@
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ConditionLinkSetMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.DiaryPlanStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.DocumentReferenceToNarrativeStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.EhrExtractMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.EncounterComponentsMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.EncounterMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ImmunizationObservationStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.MedicationStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.MessageContext;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.NonConsultationResourceMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ObservationStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ObservationToNarrativeStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.OrganizationToAgentMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.OutputMessageWrapperMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ParticipantMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.PertinentInformationObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.PractitionerAgentPersonMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.RequestStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.StructuredObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.DiagnosticReportMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.ObservationMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.SpecimenMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class TransformJsonToXml {
+
+    private static final String JSON_FILE_INPUT_PATH =
+            Paths.get("src").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/input/";
+    private static final String XML_OUTPUT_PATH =
+            Paths.get("src").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/output/";
+    private static final Logger LOGGER = Logger.getLogger(TransformJsonToXml.class.getName());
+
+    private static final FhirParseService FHIR_PARSE_SERVICE = new FhirParseService();
+
+    public static void main(String[] args) throws Exception {
+
+        var inputWrapper = getFiles();
+
+        for (int i = 0; i < inputWrapper.getJsonFileInputs().size(); i++) {
+            var jsonString = inputWrapper.getJsonFileInputs().get(i);
+            String xmlResult = mapJsonToXml(jsonString);
+            var fileName = inputWrapper.getJsonFileNames().get(i);
+            writeToFile(xmlResult, fileName);
+        }
```

Comment:
I suggest extracting the lists so you don't have to get them new each time they're needed
```suggestion
        var jsonFileInputs = inputWrapper.getJsonFileInputs();
        var jsonFileNames = inputWrapper.getJsonFileNames();
        for (int i = 0; i < jsonFileInputs.size(); i++) {
            String jsonString = jsonFileInputs.get(i);
            String xmlResult = mapJsonToXml(jsonString);
            String fileName = jsonFileNames.get(i);
            writeToFile(xmlResult, fileName);
        }
```